suite.go

```diff
@@ -68,6 +68,42 @@ type suite struct {
 	afterScenarioHandlers  []AfterScenarioHook
 }
 
+type Attachment struct {
+	Body      []byte
+	FileName  string
+	MediaType string
+}
+
+type attachmentKey struct{}
+
+func Attach(ctx context.Context, attachments ...Attachment) context.Context {
```

Comment:
The API of `Attach`/`Attachments` seems reasonable, but maybe needs a bit of documentation (as a function comment) on the expected behavior. One may think that

```go
ctx = godog.Attach(ctx, a1)
// ..... things happening here
ctx = godog.Attach(ctx, a2)
```
would add both `a1` and `a2` as attachments, though instead they would need to use something like

```go
ctx = godog.Attach(ctx, a1)
// ..... things happening here
ctx = godog.Attach(ctx, append(godog.Attachments(ctx), a2)
```

Reply:
I've improved the impl so its convenient in the situation mentioned.

This now works ..

    ctx = godog.Attach(ctx, a1)
    // ..... things happening here  
    ctx = godog.Attach(ctx, a2)